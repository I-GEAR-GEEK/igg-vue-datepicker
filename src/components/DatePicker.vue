<template>
  <DatePicker
    ref="datepicker"
    v-model="date"
    format="DD/MM/YYYY"
    :editable="false"
    :placeholder="placeholder"
    :lang="lang"
    :disabled-date="disabledDate"
    style="width: 100%"
  />
</template>

<script>
import DatePicker from 'vue2-datepicker'
import 'vue2-datepicker/index.css'
import { LANG } from '@/assets/constant/datepicker'

export default {
  components: {
    DatePicker
  },
  props: {
    value: {
      type: Date,
      default: null
    },
    placeholder: {
      type: String,
      default: ''
    },
    disabledDate: {
      type: Function,
      default: () => {}
    }
  },
  data () {
    return {
      lang: LANG,
      date: this.value
    }
  },
  watch: {
    value (val) {
      this.date = val
    },
    date (val) {
      this.$emit('input', val)
    }
  }
}
</script>
