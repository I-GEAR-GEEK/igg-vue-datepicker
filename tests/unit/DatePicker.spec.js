
import { shallowMount } from '@vue/test-utils'
import DatePicker from '@/components/DatePicker.vue'

global.Date.now = jest.fn(() => new Date('2021-07-02T00:00:00'))

describe('DatePicker.vue', () => {
  it('should match with snapshot', () => {
    const wrapper = shallowMount(DatePicker, {
      placeholder: 'placeholder'
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should match with value', async () => {
    const date = new Date()
    const wrapper = await shallowMount(DatePicker, {
      propsData: {
        value: date
      }
    })

    expect(wrapper.vm.$data.date).toEqual(date)
  })

  it('should match with on change value', async () => {
    const date = new Date('2021-07-01T00:00:00')
    const wrapper = await shallowMount(DatePicker, {
      propsData: {
        value: new Date()
      }
    })

    await wrapper.setProps({
      value: date
    })

    expect(wrapper.vm.$data.date).toEqual(date)
  })
})
