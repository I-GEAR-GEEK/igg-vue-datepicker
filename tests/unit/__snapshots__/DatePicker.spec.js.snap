// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker.vue should match with snapshot 1`] = `
<datepicker-stub defaultvalue="Mon Jul 12 2021 00:00:00 GMT+0700 (Indochina Time)" disableddate="() => {}" type="date" getclasses="function _default() {
          return [];
        }" titleformat="YYYY-MM-DD" format="DD/MM/YYYY" timetitleformat="YYYY-MM-DD" disabledtime="function _default() {
          return false;
        }" hourstep="1" minutestep="1" secondstep="1" scrollduration="100" valuetype="date" lang="[object Object]" placeholder="" clearable="true" prefixclass="mx" inputattr="[object Object]" appendtobody="true" popupstyle="[object Object]" confirmtext="OK" shortcuts="" style="width: 100%;"></datepicker-stub>
`;
